conans/client/rest/conan_requester.py

```diff
@@ -17,6 +18,15 @@
 # TODO: Fix this security warning
 logging.captureWarnings(True)
 
+class SSLContextAdapter(HTTPAdapter):
+    def init_poolmanager(self, *args, **kwargs):
+        context = create_urllib3_context()
+        kwargs['ssl_context'] = context
+        try:
+            context.load_default_certs()
```

Comment:
I would call this function before modifying `kwargs`. If anything fails, this overrides will do nothing and we will have a `HTTPAdapter` like we had before